core/src/main/java/io/questdb/cutlass/line/tcp/LineTcpParser.java

```diff
@@ -430,9 +416,15 @@ private boolean expectEntityValue(byte endOfEntityByte, long bufHi) {
     private boolean expectTableName(byte endOfEntityByte, long bufHi) {
         tagsComplete = endOfEntityByte == (byte) ' ';
         if (endOfEntityByte == (byte) ',' || tagsComplete) {
-            measurementName.of(entityLo, bufAt - nEscapedChars);
-            entityHandler = entityNameHandler;
-            return true;
+            long hi = sanitizeEnd(entityLo, bufAt - nEscapedChars);
```

Comment:
I'm not sure we have to tolerate \0 at the end of measurement name. 

From another hand ending table name with `.` is allowed in SQL in QuestDB atm:
```
create table "a."  (col int)
select * from "a."
```

and I don't see why we cannot allow ILP to use it in the table name.

`\ a` in ILP is also valid and means same as table `" a"` in SQL

I think we need to strip leading `\0` only and add protection in QuestDB engine to check table names as well as

Reply:
I could be missing something, but we do not tolerate `\0` neither at start nor at end. The `.` is only removed at start, to avoid `../table` shenanigans. At end it is allowed.

```
I'm not sure we have to tolerate \0 at the end of measurement name.
```

If you meant to say here we should allow `\0` at end - it is useless. When creating directory we use zero-terminated strings.

I'm going to remove ` ` from sanitisation though, ILP parser takes care of that already